### PR TITLE
Unhandled promise rejection - attach non-standard Error object stack info if possible

### PR DESCRIPTION
**Pull request by ospfranco**
## Summary:

This is a continuation of my [last PR](https://github.com/facebook/react-native/pull/40914) which improved the symbolication of unhandled promise rejections.

While I was developing another library I noticed I still got an error stack of the log adding and not of the error itself. The library I'm trying to debug does not throw a standard error object but rather a custom one, but it still contains the stack field. By passing this stack field to the logbox call I was able to get a better symbolicated stack trace. The exact line of the failure is not displayed but at least the correct file is.

## Test Plan:

Test any unhandled promise rejection with a non-standard error (line 23, toString must not return `[object Error]`) and see if the correct (or at least a better) stack trace is shown.

Here is the one I got before and after this change:

### CODE REVIEW

**Review by cipolleschi — APPROVED** [review]
This looks good to me, but I don't know this part of the framework as well as other people, so I'll let them have another round of review.

**Comment by ospfranco** [comment]
@javache  reviewed the last PR

File: packages/react-native/Libraries/promiseRejectionTrackingOptions.js
Comment (joe-sam) [inline packages/react-native/Libraries/promiseRejectionTrackingOptions.js:1]: I might be mistaken but shouldn't the _rejection.stack_ unwinding be done optimally before the final _JSON.stringify((rejection: $FlowFixMe))_?

Comment (ospfranco) [reply packages/react-native/Libraries/promiseRejectionTrackingOptions.js:1]: No, I don't think so, Stringify does not unwind the stack or show it in the logbox

**Pull request closed without merging**